<?xml version="1.0" encoding="utf-8"?>
<launch>
  <!-- Load Parameter -->
  <param name="yolov5_path" value="$(find yolov5_ros)/yolov5"/>
  <param name="use_cpu" value="true" />

  <!-- Start usb_cam node to capture camera images -->
  <node pkg="usb_cam" type="usb_cam_node" name="usb_cam" output="screen">
    <param name="video_device" value="/dev/video0" />
    <param name="image_width" value="640" />
    <param name="image_height" value="480" />
    <param name="pixel_format" value="yuyv" />
    <param name="io_method" value="mmap" />
    <param name="camera_frame_id" value="camera_color_frame" />
  </node>

  <!-- Start yolov5 and ros wrapper -->
  <node pkg="yolov5_ros" type="yolo_v5.py" name="yolov5_ros" output="screen">
    <param name="weight_path" value="$(find yolov5_ros)/weights/best.pt"/>
    <param name="image_topic" value="/usb_cam/image_raw" />
    <param name="pub_topic" value="/yolov5/BoundingBoxes" />
    <param name="camera_frame" value="camera_color_frame"/>
    <param name="conf" value="0.5" />
  </node>

  <!-- Start tts_subscriber node for text-to-speech -->
  <node pkg="yolov5_ros" type="tts_subscriber.py" name="tts_subscriber" output="screen"/>

  <!-- Start voice command publisher -->
  <node pkg="yolov5_ros" type="voice_command_publisher.py" name="voice_command_publisher" output="screen"/>
</launch>

